Makefile

```diff
@@ -4,43 +4,32 @@
 mkfile_path := $(abspath $(lastword $(MAKEFILE_LIST)))
 mkfile_dir := $(dir $(mkfile_path))
 
-export TARGET_ARCH ?= $(shell uname -m | sed s/x86_64/amd64/ | sed s/aarch64/arm64/)
-export TARGET_TRIPLE ?= $(shell command -v rustc 1>/dev/null && rustc -vV | sed -n 's|host: ||p')
-
-# High-level settings that ultimately get passed down to build-specific targets.
-export APP_FULL_NAME ?= Agent Data Plane
-export APP_SHORT_NAME ?= data-plane
-export APP_IDENTIFIER ?= adp
-export APP_GIT_HASH ?= $(shell git rev-parse --short HEAD 2>/dev/null || echo not-in-git)
-export APP_VERSION ?= $(shell cat bin/agent-data-plane/Cargo.toml | grep -E "^version = \"" | head -n 1 | cut -d '"' -f 2)
+# High-level settings that ultimately get passed down to build-specific targets and other spots.
+export TARGET_ARCH := $(shell uname -m | sed s/x86_64/amd64/ | sed s/aarch64/arm64/)
```

Comment:
For many of the variables here, we're simply changing from `=?` to `:=`, and you may ask "why? what's that doing?".

In Make, variables defined via `=` are "recursively expanded", which means they're expanded every time they're referenced. `?=` is just an extension of this, where the variable isn't set to the right-hand side if it's already defined.. but it's still recursively expanded.

What this means is that it can end up running these commands we're using to get the value _many times_ when parsing the Makefile, to the point where `make` commands are noticeably slow.

For these commands where we really only need to set the value once, we use "simply expanded" variables, denoted by the use of `:=`.